### PR TITLE
[FEATURE] Remove redundant param from FileRenderer render methods

### PR DESCRIPTION
**Pull request by helsner**
Resolves: #2867

### CODE REVIEW

**Comment by helsner** [comment]
Hi @simonschaufi ,
i got some brain fog here.. shouldn't it work to check for the interface as well? Because that would be the smartest
Other than that i could manually check for alle object types, but this would no include the custom ones and also feels superfluous if we can ask for the interface..

Secondly i don't know if we should cover the case someone had a custom FileRenderer that implements the interface.
That would be a second rule i guess, see overall comments in the PR

**Comment by simonschaufi** [comment]
First of all, please make sure all tests are :heavy_check_mark:

File: stubs/TYPO3/CMS/Core/Resource/Rendering/AudioTagRenderer.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+     * @param array $options controls = TRUE/FALSE (default TRUE), autoplay = TRUE/FALSE (default FALSE), loop = TRUE/FALSE (default FALSE)
+     * @param bool $usedPathsRelativeToCurrentScript See $file->getPublicUrl()
+     * @return string
+     */
+    public function render(FileInterface $file, $width, $height, array $options = [], $usedPathsRelativeToCurrentScript = false)
+    {
+        return '';
+    }
+
+}
```
Comment (simonschaufi) [inline stubs/TYPO3/CMS/Core/Resource/Rendering/AudioTagRenderer.php:30]: New line missing. i wonder why your phpstorm doesn't do that

File: stubs/TYPO3/CMS/Core/Resource/Rendering/FileRendererInterface.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+
```
Comment (simonschaufi) [inline stubs/TYPO3/CMS/Core/Resource/Rendering/FileRendererInterface.php:3]: only one empty line

File: stubs/TYPO3/CMS/Core/Resource/Rendering/VideoTagRenderer.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+     * @param array $options controls = TRUE/FALSE (default TRUE), autoplay = TRUE/FALSE (default FALSE), loop = TRUE/FALSE (default FALSE)
+     * @param bool $usedPathsRelativeToCurrentScript See $file->getPublicUrl()
+     * @return string
+     */
+    public function render(FileInterface $file, $width, $height, array $options = [], $usedPathsRelativeToCurrentScript = false)
+    {
+        return '';
+    }
+
+}
```
Comment (simonschaufi) [inline stubs/TYPO3/CMS/Core/Resource/Rendering/VideoTagRenderer.php:30]: New line missing.

File: stubs/TYPO3/CMS/Core/Resource/Rendering/VimeoRenderer.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+     * @param array $options controls = TRUE/FALSE (default TRUE), autoplay = TRUE/FALSE (default FALSE), loop = TRUE/FALSE (default FALSE)
+     * @param bool $usedPathsRelativeToCurrentScript See $file->getPublicUrl()
+     * @return string
+     */
+    public function render(FileInterface $file, $width, $height, array $options = [], $usedPathsRelativeToCurrentScript = false)
+    {
+        return '';
+    }
+
+}
```
Comment (simonschaufi) [inline stubs/TYPO3/CMS/Core/Resource/Rendering/VimeoRenderer.php:30]: New line missing.

File: stubs/TYPO3/CMS/Core/Resource/Rendering/YoutubeRenderer.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+     * @param array $options controls = TRUE/FALSE (default TRUE), autoplay = TRUE/FALSE (default FALSE), loop = TRUE/FALSE (default FALSE)
+     * @param bool $usedPathsRelativeToCurrentScript See $file->getPublicUrl()
+     * @return string
+     */
+    public function render(FileInterface $file, $width, $height, array $options = [], $usedPathsRelativeToCurrentScript = false)
+    {
+        return '';
+    }
+
+}
```
Comment (simonschaufi) [inline stubs/TYPO3/CMS/Core/Resource/Rendering/YoutubeRenderer.php:30]: New line missing.

File: tests/Rector/v12/v0/typo3/RemoveRelativeToCurrentScriptArgumentsRector/Fixture/fixture.php.inc [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+$foo = $audioTagRenderer->render($file, $width, $height, $options, $relative);
+$videoTagRenderer = GeneralUtility::makeInstance(VideoTagRenderer::class);
+$foo = $videoTagRenderer->render($file, $width, $height, $options, $relative);
+$vimeoRenderer = GeneralUtility::makeInstance(VimeoRenderer::class);
+$foo = $vimeoRenderer->render($file, $width, $height, $options, $relative);
+$youtubeRenderer = GeneralUtility::makeInstance(YoutubeRenderer::class);
+$foo = $youtubeRenderer->render($file, $width, $height, $options, $relative);
+
+
+$image = GeneralUtility::makeInstance(\TYPO3\CMS\Backend\Backend\Avatar\Image::class);
```
Comment (simonschaufi) [inline tests/Rector/v12/v0/typo3/RemoveRelativeToCurrentScriptArgumentsRector/Fixture/fixture.php.inc:36]: import namespace

File: tests/Rector/v12/v0/typo3/RemoveRelativeToCurrentScriptArgumentsRector/Fixture/fixture.php.inc [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+$urlTwo = $image->getUrl();
+
+?>
+-----
+<?php
+
+namespace Ssch\TYPO3Rector\Tests\Rector\v12\v0\typo3\RemoveRelativeToCurrentScriptArgumentsRector\Fixture;
+
+use TYPO3\CMS\Backend\Backend\Avatar\Image;
+use Ssch\TYPO3Rector\Tests\Rector\v12\v0\typo3\RemoveRelativeToCurrentScriptArgumentsRector\Source\TestFileRendering;
```
Comment (simonschaufi) [inline tests/Rector/v12/v0/typo3/RemoveRelativeToCurrentScriptArgumentsRector/Fixture/fixture.php.inc:47]: order is wrong

**Comment by helsner** [comment]
> First of all, please make sure all tests are ✔️

regarding the tests i need some feedback of you.
1. do you know about the interface handling? because i would like to do it with the interface check instead of each renderer, as this is cleaner and would also cover custom cases.
2. do you think we should create a second rector rule to check for classes that implement the interface and the render function to then remove the argument from there?

Basically the tests are there to show the TODOs to discuss

**Comment by sabbelasichon** [comment]
@helsner Could you have a look at what i have done

**Comment by helsner** [comment]
I already did, looks good.
What i'm wondering though is why the Interface check did not work in my case??

but you handled everything i thought about, so also the custom rendering stuff.
in case you got nothing else we can merge this. As PR author i cannot approve, but do so by this message 👍
